finesse/gui/interferometer_monitor.py

```diff
@@ -0,0 +1,204 @@
+        self._laser_current_box.setAlignment(Qt.AlignmentFlag.AlignCenter)
+
+        self._poll_light = LEDIcon.create_poll_icon()
+
+        layout = self._create_controls()
+
+        self._poll_light.setSizePolicy(self._laser_current_box.sizePolicy())
+
+        self.setLayout(layout)
+
+    def _create_controls(self) -> QGridLayout:
+        """Creates the widgets for the EM27 properties.
+
+        Returns:
+            QGridLayout: The layout with the widgets.
+        """
+        layout = QGridLayout()
+
+        # Add labels for properties to monitor
+        for i, label in enumerate(self._prop_labels):
+            layout.addWidget(QLabel(label), i, 0)
+
+        # Add boxes to show monitored properties' values
+        layout.addWidget(self._psf27_temp_box, 0, 1)
+        layout.addWidget(self._cryo_temp_box, 1, 1)
+        layout.addWidget(self._bb_hum_box, 2, 1)
+        layout.addWidget(self._src_temp_box, 3, 1)
+        layout.addWidget(self._aux_volt_box, 4, 1)
+        layout.addWidget(self._aux_current_box, 5, 1)
+        layout.addWidget(self._laser_current_box, 6, 1)
+        layout.addWidget(self._poll_light, 7, 1)
+
+        return layout
+
+    def set_psf27_temp(self, val: tuple[str, float]) -> None:
+        """Sets the PSF27 temperature text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._psf27_temp_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_cryo_temp(self, val: tuple[str, float]) -> None:
+        """Sets the cryo temperature text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._cryo_temp_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_bb_hum(self, val: tuple[str, float]) -> None:
+        """Sets the blackbody humidity text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._bb_hum_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_src_temp(self, val: tuple[str, float]) -> None:
+        """Sets the source temperature text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._src_temp_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_aux_volt(self, val: tuple[str, float]) -> None:
+        """Sets the AUX voltage text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._aux_volt_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_aux_current(self, val: tuple[str, float]) -> None:
+        """Sets the AUX current text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._aux_current_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def set_laser_current(self, val: tuple[str, float]) -> None:
+        """Sets the laser current text box.
+
+        Args:
+            val: value polled from server
+        """
+        self._laser_current_box.setText(f"{val[0]:.6f} {val[1]}")
+
+    def poll_server(self) -> None:
```

Comment:
It seems like you're not calling this from anywhere atm. I'd make it actually work: add a `QTimer` to invoke it every 2s or something.